packages/framework-core/src/booster-event-dispatcher.ts

```diff
@@ -89,7 +92,18 @@ export class BoosterEventDispatcher {
         logger.debug(`No event-handlers found for event ${eventEnvelope.typeName}. Skipping...`)
         continue
       }
-      const eventClass = config.events[eventEnvelope.typeName]
+      const eventClass = BoosterEventDispatcher.eventMetadata(eventEnvelope, config)
+      if (!eventClass) {
+        throw new NotFoundError(
```

Comment:
@gonzalojaubert  Does it make sense to use the global dispatcher here to track this error?